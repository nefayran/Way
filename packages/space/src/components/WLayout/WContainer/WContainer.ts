import { computed, h, defineComponent } from "vue";

export default defineComponent({
  name: "WContainer",

  props: {
    size: {
      type: [Boolean, String],
      default: "fluid",
    },
    tag: {
      type: String,
      default: "div",
    },
  },

  setup(props, { slots }) {
    const classes = computed(() => {
      const classList: any[] = [];

      classList.push({
        [`container-${props.size}`]: props.size,
      });

      return classList;
    });

    return () =>
      h(
        props.tag,
        {
          class: ["w-container", classes.value],
        },
        slots.default?.()
      );
  },
});
