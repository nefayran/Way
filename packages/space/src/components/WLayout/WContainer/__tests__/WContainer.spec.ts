import { mount } from "@vue/test-utils";
import WContainer from "../WContainer";

describe("WContainer.ts", () => {
  test("should have an w-container class", () => {
    const wrapper = mount(WContainer, {
      propsData: {
        size: "lg",
      },
    });
    expect(wrapper.classes()).toContain("w-container");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
