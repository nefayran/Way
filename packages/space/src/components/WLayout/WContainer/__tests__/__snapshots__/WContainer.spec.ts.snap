// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WContainer.ts should have an w-container class 1`] = `<div class="w-container container-lg"></div>`;
